dynamic_programming/catalan_numbers.nim

```diff
@@ -113,7 +114,7 @@ when isMainModule:
     test "We can compute up to the thirty-seventh Catalan number iteratively":
       let limit = UpperLimit
       let catalanNumbersSeq = catalanNumbers(limit)
-      for index in 0 .. limit:
+      for index in 0 ..< limit:
         check catalanNumbersSeq[index] == CatalanNumbersList[index]
```

Comment:
This is not a syntax issue. Have you run the tests? I am pretty sure it worked by including the upper bound.

Reply:
This algorithm does not have plenty of tests.
I will try to imagine some easy examples not to hard to compute by hand for better verification after this P.R.

Reply:
> This is not a syntax issue. Have you run the tests? I am pretty sure it worked by including the upper bound.

- [`limit`](https://github.com/TheAlgorithms/Nim/blob/c38d3bc59bf3bb79ac6ca1d58d73d531d4d497ae/dynamic_programming/catalan_numbers.nim#L115) is set to [`36`](https://github.com/TheAlgorithms/Nim/blob/c38d3bc59bf3bb79ac6ca1d58d73d531d4d497ae/dynamic_programming/catalan_numbers.nim#L80),
- [`CatalanNumbersList`](https://github.com/TheAlgorithms/Nim/blob/c38d3bc59bf3bb79ac6ca1d58d73d531d4d497ae/dynamic_programming/catalan_numbers.nim#L85) has 36 elements, the last element is `CatalanNumbersList[35]`
- [`catalanNumbersSeq`](https://github.com/TheAlgorithms/Nim/blob/c38d3bc59bf3bb79ac6ca1d58d73d531d4d497ae/dynamic_programming/catalan_numbers.nim#L116) also has 36 elements (note `<` in [catalanNumbers](https://github.com/TheAlgorithms/Nim/blob/c38d3bc59bf3bb79ac6ca1d58d73d531d4d497ae/dynamic_programming/catalan_numbers.nim#L37)).
- without this change I get `Unhandled exception: index 36 not in 0 .. 35 [IndexDefect]` (both by executing the command `nim test` or by running this file directly e.g. `nim r catalan_numbers.nim`).

Reply:
I was thinking about Viterbi. Catalan Numbers does have plenty of tests, which made us notice this bug (probably a merge error).